### PR TITLE
UI - build optimize

### PR DESCRIPTION
**Pull request by meirish**
When doing development, the build times have crept up over various additions, updates, and development cycles. This removes some things we don't need in development from the build. This also updates `ember-cli` to 3.5.x which has a new version of Broccoli that uses the OS tmp dir and has other performance improvements.

Additionally, this removes ember-cli-favicon which fixed a bug where the linked favicon were being re-loaded on every page navigation in the application.

Here are some unscientific numbers illustrating the before and after - I did each of these a couple of times on the before I took the best time, the after I took the worst one.
<details>
<summary>Output of the ember dev server</summary>
Cold start:
Before:

```
Build successful (49212ms) – Serving on http://localhost:4200/ui/

Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
broccoli-persistent-filter:EslintValid... (2) | 9993ms (4996 ms)
ember-auto-import-analyzer (4)                | 6630ms (1657 ms)
Babel: vault (2)                              | 4975ms (2487 ms)
Package /assets/vendor.js (1)                 | 3763ms
```
after:

```
Build successful (15382ms) – Serving on http://localhost:4200/ui/

Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
SassCompiler (1)                              | 4317ms
ember-auto-import-analyzer (2)                | 2004ms (1002 ms)
Babel: vault (1)                              | 1829ms
Rollup (3)                                    | 1436ms (478 ms)
Bundler (1)                                   | 1318ms
```

Template file change:
Before:
```
Build successful (5360ms) – Serving on http://localhost:4200/ui/

Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
SassCompiler (1)                              | 1510ms
Packaged Application Javascript (1)           | 892ms
broccoli-persistent-filter:Autoprefixe... (1) | 426ms
ember-auto-import-analyzer (4)                | 332ms (83 ms)
Pod Templates (1)                             | 292ms
```

After:
```
Build successful (1004ms) – Serving on http://localhost:4200/ui/

Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
Pod Templates (1)                             | 207ms
Processed Application and Dependencies (2)    | 116ms (58 ms)
Babel: vault (1)                              | 96ms
Packaged Application Javascript (1)           | 94ms
Package /assets/vendor.js (1)                 | 74ms
broccoli-persistent-filter:TemplateCom... (6) | 64ms (10 ms)
ember-auto-import-analyzer (2)                | 61ms (30 ms)
```

JS file change:

Before:

```
Build successful (6092ms) – Serving on http://localhost:4200/ui/

Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
SassCompiler (1)                              | 1494ms
Packaged Application Javascript (1)           | 755ms
broccoli-persistent-filter:Autoprefixe... (1) | 398ms
Pod Templates (1)                             | 364ms
Concat: App Tests (1)                         | 343ms
SimpleConcatConcat (3)                        | 325ms (108 ms)
```

After:
```
Build successful (1189ms) – Serving on http://localhost:4200/ui/

Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
Packaged Application Javascript (1)           | 300ms
Pod Templates (1)                             | 189ms
Package /assets/vendor.js (1)                 | 165ms
Processed Application and Dependencies (2)    | 134ms (67 ms)
ember-auto-import-analyzer (2)                | 63ms (31 ms)
```

SCSS file change:
Before: 
```
Build successful (4683ms) – Serving on http://localhost:4200/ui/

Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
SassCompiler (1)                              | 1137ms
Processed Application and Dependencies (2)    | 449ms (224 ms)
broccoli-persistent-filter:Autoprefixe... (1) | 404ms
SimpleConcatConcat (3)                        | 395ms (131 ms)
Pod Templates (1)                             | 313ms
TemplateLinter (1)                            | 256ms
```

After:
```
Build successful (4203ms) – Serving on http://localhost:4200/ui/

Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
SassCompiler (1)                              | 3353ms
Pod Templates (1)                             | 216ms
```

After with node-sass:

```
Build successful (1502ms) – Serving on http://localhost:4200/ui/

Slowest Nodes (totalTime => 5% )              | Total (avg)
----------------------------------------------+---------------------
SassCompiler (1)                              | 708ms
Pod Templates (1)                             | 179ms
Package /assets/vendor.js (1)                 | 95ms
Processed Application and Dependencies (2)    | 87ms (43 ms)

file changed styles/components/auth-form.scss
```
</details>

I'll leave some notes on the commit about what we're cutting out to get the speed boost.

### CODE REVIEW

**Comment by madalynrose** [comment]
omg these numbers! 💨 💨 💨 💨 💨

File: ui/app/styles/core.scss [hunk 1]
```diff
@@ -1,83 +1,82 @@
 // Start with Structure & Bulma variables as a foundation
-@import "./utils/colors";
-@import "./utils/spacing";
-@import "bulma/sass/utilities/initial-variables";
+@import './utils/colors';
```
Comment (meirish) [inline ui/app/styles/core.scss]: lol prettier strikes again

File: ui/app/styles/core.scss [hunk 2]
```diff
@@ -1,83 +1,82 @@
...
 // Utils
-@import "./utils/mixins";
-@import "./utils/animations";
+@import './utils/mixins';
+@import './utils/animations';
 
 // Bring in the rest of Bulma
-@import "bulma/bulma";
-@import "bulma/switch";
-@import "bulma/bulma-radio-checkbox";
```
Comment (meirish) [inline ui/app/styles/core.scss]: This was the only change here - this was a vendor CSS file and importing it in sass caused a warning - I moved it to `ember-cli-build.js` and used `app.import` to add it to the vendor CSS file.

File: ui/app/styles/core/forms.scss [hunk 3]
```diff
@@ -184,16 +184,16 @@ label {
...
+.select:not(.is-multiple)::after,
+.select:not(.is-multiple)::before {
   border-color: $black;
   border-width: 2px;
   margin-top: 0;
   transform: translateY(30%) rotate(-45deg);
 }
 
 .select:not(.is-multiple)::before {
-  @extend .select:not(.is-multiple)::after;
```
Comment (meirish) [inline ui/app/styles/core/forms.scss]: Dart Sass didn't like this - we didn't end up using that, but I also thought it wasn't super necessary so I added the selector above.

File: ui/ember-cli-build.js [hunk 4]
```diff
@@ -33,9 +18,20 @@ module.exports = function(defaults) {
       plugins: ['transform-object-rest-spread'],
     },
     'ember-cli-babel': {
-      includePolyfill: true,
+      includePolyfill: isProd,
```
Comment (meirish) [inline ui/ember-cli-build.js]: This may need to be `isProd || isTest` so that IE testing works when we get there.

File: ui/ember-cli-build.js [hunk 5]
```diff
@@ -33,9 +18,20 @@ module.exports = function(defaults) {
...
+    },
+    hinting: isTest,
```
Comment (meirish) [inline ui/ember-cli-build.js]: The build server won't show eslint errors anymore - but with the husky PR we'll get them in the pre-commit hook.

File: ui/ember-cli-build.js [hunk 6]
```diff
@@ -33,9 +18,20 @@ module.exports = function(defaults) {
...
+    tests: isTest,
+    sourcemaps: {
+      enabled: false,
+    },
+    sassOptions: {
+      sourceMap: false,
```
Comment (meirish) [inline ui/ember-cli-build.js]: This was the thing I wasn't sure if people were using. You _can_ configure your chrome workspace to directly edit SCSS files in the dev tools, but was anyone working that way?

File: ui/ember-cli-build.js (hunk 5 above)
Comment (noelledaley) [reply ui/ember-cli-build.js]: that seems fine, since we should also still get eslint warnings in our editors, correct?

Comment (meirish) [reply ui/ember-cli-build.js]: Yep! You'd only see them before if you were looking at the app build output (which can scroll by pretty fast depending on what you're doing). Editor integration + husky hooks + tests failing lint errors will still catch things.

**Pull request merged**